rust/src/dcerpc/dcerpc.rs

```diff
@@ -982,40 +944,38 @@ impl DCERPCState {
         // Check if header data was complete. In case of EoF or incomplete data, wait for more
         // data else return error
         if self.bytes_consumed < DCERPC_HDR_LEN.into() && input_len > 0 {
```

Comment:
I find this use of a `bytes_consumed` in the state a bit confusing, not that we return incomplete if we don't have the full header. Why do we need something in the state for it?

Reply:
> Why do we need something in the state for it?

This is for when we have a complete header but the rest of the fragment is incomplete. This helps us track whether we have parsed the header for this PDU or not.
I have tried using `header.is_none()` but its a bit tricky to reset the header.
Does this make sense to you?

Reply:
can you use the incomplete logic? First use incomplete to get the full header, parse it. Then you know the rest of the len, so just return incomplete if you don't have that yet?

Reply:
It seems like a simple problem to solve but I failed at making this work.

Since you insisted on this, I checked other protocol parsers and I think perhaps there's a problem with the design of dcerpc parser as header is a part of the `State` struct instead of `Transaction` so, if I try to get rid of `bytes_consumed` from state and decipher whether the header has been parsed or not, I keep hitting edge cases or miss information in logger for certain fields.
An easy fix is to set the required fields in the tx from the header and use them wherever needed. Doing this for now but perhaps the design needs to be reworked..

Reply:
I think this comes from the conversion from C, where there was no or minimal tx support. So it seems to make sense to further clean this up.